src/software/SfM/python/external_features_demo/kornia_demo.py

```diff
@@ -0,0 +1,142 @@
+      for x, y in keypoints.numpy():
+        feat.write(f"{x} {y} 1.0 0.0\n")
+    
+    with open(os.path.join(args.matches, f"{basename}.desc"), 'wb') as desc:
+      desc.write(len(keypoints).to_bytes(8, byteorder='little'))
+      desc.write(((features.descriptors.numpy() + 1) * 0.5 * 255).round(0).astype(np.ubyte).tobytes())
+
+# // M A T C H I N G /////////////////////////////////////////////////
+def featureMatching():
+  print('Matching DISK features with LightGlue...')
+  views = loadJSON()
+  putative_matches = []
+  for image1_index, image2_index in (np.loadtxt(args.pair_list, dtype=np.int32) if args.pair_list != None else np.asarray([*combinations(views, 2)], dtype=np.int32)):
+    keyp1, desc1, scor1 = loadFeatures(views[image1_index])
+    keyp2, desc2, scor2 = loadFeatures(views[image2_index])
+    
+    lafs1 = K.feature.laf_from_center_scale_ori(keyp1[None], 96 * torch.ones(1, len(keyp1), 1, 1, device=device))
+    lafs2 = K.feature.laf_from_center_scale_ori(keyp2[None], 96 * torch.ones(1, len(keyp2), 1, 1, device=device))
+    
+    dists, idxs = lightglue(desc1, desc2, lafs1, lafs2)
+    
+    putative_matches.append([image1_index, image2_index, idxs.cpu().numpy().astype(np.int32)])
+  
+  print('Saving putative matches...')
+  saveMatches(putative_matches)
+
+if __name__ == '__main__':
+
+  parser = ArgumentParser()
+  parser.add_argument('--images', type=str, required=True, help='Path to the image directory')
+  parser.add_argument('--max_resolution', type=int, default=1024, help='Max image resolution')
+  parser.add_argument('--max_features', type=int, default=2048, help='Max number of features to extract')
+  parser.add_argument('--matches', type=str, required=True, help='Path to the matches directory')
+  parser.add_argument('--sfm_data', type=str, required=True, help='Path to the sfm_data file')
+  parser.add_argument('--pair_list', type=str, help='Path to the pair file')
+  parser.add_argument('--output', type=str, required=True, help='Path to the output matches file')
+  parser.add_argument('--force_cpu', action='store_true', help='Force device to CPU')
+  # DISK
+  parser.add_argument('--window_size', type=int, default=5, help='DISK Non Maximum Suppression (NMS) radius (Must be odd)')
+  parser.add_argument('--score_threshold', type=float, default=0.0, help='DISK keypoint detector confidence threshold')
+  # LightGlue
+  parser.add_argument('--n_layers', type=int, default=9, help='LightGlue layers')
+  parser.add_argument('--depth_confidence', type=float, default=0.95, help='LightGlue early stopping (-1 - disable)')
+  parser.add_argument('--width_confidence', type=float, default=0.99, help='LightGlue point pruning (-1 - disable)')
+  parser.add_argument('--filter_threshold', type=float, default=0.2, help='LightGlue match threshold')
+  args = parser.parse_args()
+  
+  image_list = []
+  {image_list.extend(glob(os.path.join(args.images, f'*.{file_extension}'))) for file_extension in ['jpg', 'jpeg', 'png']}
```

Comment:
Maybe convert file_extension to lower case. Got the issue on my machine and some dataset since my images were .JPG (upper case)